Dockerfile

```diff
@@ -34,7 +47,14 @@ COPY lib /app/lib
 
 # Assets
 COPY vendor/assets /app/vendor/assets
-RUN echo "takes 5 minute" && ./bin/decode_dot_env .env && RAILS_ENV=production PRECOMPILE=1 bundle exec rake assets:precompile
+RUN echo "takes 5 minute" && ./bin/decode_dot_env .env && RAILS_ENV=production PRECOMPILE=1 bundle exec rake assets:precompile 2>/dev/null
+
+# run as non-root, but allow it to access docker, use id so kubernetes non-root check passes
+RUN groupadd -r samson \
+  && useradd -r -g samson -d /app -u 1000 samson \
+  && chown -R samson:samson /app \
```

Comment:
This will duplicate the entire source code layer unfortunately, check out the output of `docker history samson`.